flocker/node/functional/test_deploy.py

```diff
@@ -360,11 +360,13 @@ def inspect_application(_):
             deferred_list = docker_client.list()
 
             def app_memory(units):
-                unit = units.pop()
-                self.assertEqual(unit.mem_limit, EXPECTED_MEMORY_LIMIT)
-                return deferred_list
-
+                checked = [
+                    self.assertEqual(unit.mem_limit, EXPECTED_MEMORY_LIMIT)
+                    for unit in units if unit.name == application_name
+                ]
```

Comment:
:question: `list(generator expression)` instead of `[list comprehension]` is a good habit to get in to.  Generator expressions don't leak the loop variable into the enclosing scope - and that's preferable almost all the time.  Make list comprehensions the exception rather than the default.